backend/api/v1/sql_service.go

```diff
@@ -354,6 +361,38 @@ func (s *SQLService) postExport(ctx context.Context, activity *store.ActivityMes
 	return nil
 }
 
+func doEncrypt(data []byte, request *v1pb.ExportRequest) ([]byte, error) {
+	var b bytes.Buffer
+	fzip := io.Writer(&b)
+
+	zipw := zip.NewWriter(fzip)
+
+	filename := fmt.Sprintf("export.%s", strings.ToLower(request.Format.String()))
+	var w io.Writer
+
+	if request.Password != "" {
+		withPassword, err := zipw.Encrypt(filename, request.Password)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create encrypt export file")
+		}
+		w = withPassword
+	} else {
+		withoutPassword, err := zipw.Create(filename)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create export file")
+		}
+		w = withoutPassword
+	}
+
+	_, err := io.Copy(w, bytes.NewReader(data))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to write export file")
+	}
+	zipw.Close()
```

Comment:
Should we use defer?

Reply:
Add defer to ensure the close. BTW I found need to manually call the `Close` too, otherwise cannot generate the correct zip file

Reply:
It seems that the official doc calls `Flush()`.

```go
func main() {
    ...
	zipw := zip.NewWriter(fzip)
	defer zipw.Close()
    ...
	_, err = io.Copy(w, bytes.NewReader(contents))
	if err != nil {
		log.Fatal(err)
	}
	zipw.Flush()
}
```

Reply:
the `Flush` doesn't work either..